lib/web_ui/lib/src/engine/canvaskit/picture.dart

```diff
@@ -24,14 +30,38 @@ class CkPicture extends ManagedSkiaObject<SkPicture> implements ui.Picture {
   @override
   int get approximateBytesUsed => 0;
 
+  /// Whether the picture has been disposed of.
+  /// 
+  /// This is indended to be used in tests and assertions only.
+  bool get debugIsDisposed => _isDisposed;
+
+  /// This is set to true when [dispose] is called and is never reset back to
+  /// false.
+  /// 
+  /// This extra flag is necessary on top of [rawSkiaObject] because
+  /// [rawSkiaObject] being null does not indicate permanent deletion. See
+  /// similar flag [SkiaObjectBox.isDeletedPermanently].
+  bool _isDisposed = false;
+
   @override
   void dispose() {
+    if (_isDisposed) {
+      return;
+    }
+    if (Instrumentation.enabled) {
```

Comment:
Guard so this doesn't end up in release?

Reply:
Potentially. Although as of today, the only effect is to eliminate any potentially expensive logic that might be happening in the instrumentation.